### PR TITLE
Ft cloud post

### PR DESCRIPTION
**Pull request by Melvin1Atieno**
### what this pr does
1. Adds a post on cloud and cloud computing.
### relevant links
[post](https://github.com/Melvin1Atieno/zegetech.github.io/blob/ft-cloud-post/_posts/2018-11-12-cloud-cloud-computing.md)

### CODE REVIEW

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+layout: blog
+title: The cloud and cloud  computing
+categories: developer
+author: Melvin Atieno
+blog-image: cloud-computing/cloud.jpg
+intro: What is the cloud? Who made the cloud? Where is the cloud? How does it work? Why is it indestructible? What is cloud computing? What other forms of computing are there? These are among of the questions this blog intends to answer.
+
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md]: And mother asked, "what happens when it rains?"

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+There are other types of computing such as [Distributed Computing](https://en.wikipedia.org/wiki/Distributed_computing),[Grid computing](https://en.wikipedia.org/wiki/Grid_computing),[Utility Computing](http://en.wikipedia.org/wiki/Utility_computing), and [Cluster Computing](http://en.wikipedia.org/wiki/Computer_cluster).
+
+Cloud services are delivered in a number of ways; Infrastructure as a service(IaaS), Platform as a service(PaaS) and Software as a service(SaaS). Here is a picture of resources you would traditionally manage.
+
+![Traditional IT model](/assets/images/blog/cloud-computing/traditional-IT.png){:class="img-responsive center"}
+
+### IaaS
+
+IaaS stands for Infrastructure as a service. With this service model, what you get, in simple terms, is a data center. Some aspects of computing are managed for  you by your service provider.
+
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md:45]: Please mention VPS, as that is waht is used mainly in marketing IaaS

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Here's a picture.
+![IaaS model](/assets/images/blog/cloud-computing/Iaas.png){:class="img-responsive center"}
+
+
+
+
+From the image, you can see that  networking, storage, servers, virtualization and part of the Operating system are provided as a service.
+Examples Iaas providers include; [Digital Ocean](https://www.digitalocean.com/),[linode](https://www.linode.com/), [AWS](https://aws.amazon.com/),[vultr](https://www.vultr.com/).
+
+Say you want a Linux based server. You will go to any of the above sites. Provision yourself a virtual machine and that's it. You can now deploy whatever application you want to the machine/server, in the cloud. You can make your application accessible to anyone over the internet.
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md]: virtual machine or VPS (virtual private server)

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+IaaS stands for Infrastructure as a service. With this service model, what you get, in simple terms, is a data center. Some aspects of computing are managed for  you by your service provider.
+
+Here's a picture.
+![IaaS model](/assets/images/blog/cloud-computing/Iaas.png){:class="img-responsive center"}
+
+
+
+
+From the image, you can see that  networking, storage, servers, virtualization and part of the Operating system are provided as a service.
+Examples Iaas providers include; [Digital Ocean](https://www.digitalocean.com/),[linode](https://www.linode.com/), [AWS](https://aws.amazon.com/),[vultr](https://www.vultr.com/).
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md]: add a few more popular ones like OVH and also aws it very wide, but their VPS service is called lightsail.

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+Paas stands for Platform as a Service. The service provider provides you with a pre-configured infrastructure.PaaS includes middleware, runtimes, development, testing and deployment tools.
+Here's a picture;
+
+![PaaS model](/assets/images/blog/cloud-computing/paas.png){:class="img-responsive center"}
+
+
+
+All you provide is code for your application and the data.
+Example of Paas providers are [Heroku](https://www.heroku.com/), [Go daddy](https://uk.godaddy.com), etc
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md]: Heroku is clear but godaddy is confusing. Please link to the exact product that is a PaaS. Also mention BaaS as well as opensource solution for running PaaS. FIrebase is also an example. List others here as well.

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+### SaaS
+Software as a service. The service provider fully manages the whole stack. It is an application hosted in a public domain therefore accessible to everyone.
+Here's a picture;
+
+![SaaS model](/assets/images/blog/cloud-computing/saas.png){:class="img-responsive center"}
+
+The only thing you manage is your own space. Your own environment where you have your own data.
+
+A good example of a Software as a service is Gmail provided by Google.
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md]: Please list and link to other examples. Even our own product MPAYER isa SaaS. Look for a list over the internet of Popular SaaS products and mention that zegetech has great speciality in building SaaS products and we understand this very clearly.

File: _posts/2018-11-12-cloud-cloud-computing.md (hunk 2 above)
Comment (zegetech) [reply _posts/2018-11-12-cloud-cloud-computing.md:45]: Also add Kubernetes and containerisatio infrastructure and where that fits into the cloud. is it IaaS, PaaS etc. THen give examples of providers

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+
+## Essential characteristics
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md]: Essential characteristics of what exactly. If refering to xters of cloud computing, then this should come first i should think, not at the bottom.

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+2. **Broad Network access.** Resources are available for access from a wide range of devices, such as tablets, PCs, Macs, and smartphones. These resources are also accessible from a wide range of locations that offer online access.
+
+3. **Resource pooling.** Different physical and virtual resources dynamically assigned and reassigned according to consumer demand. There is a sense of location independence in that the customer generally has no control or knowledge over the exact location of the provided resources but may be able to specify location at a higher level of abstraction (e.g country, state, or datacenter). 
+
+4. **Rapid elasticity.** Scalable provisioning, or the ability to provide scalable services. To the consumer, the capabilities available for provisioning often appear to be unlimited and can be appropriated in any quantity at any time. 
+
+5. **Measured service.** Usage can be monitored, controlled, and reported, providing transparency for both the provider and consumer of the utilized service.
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md]: Need to add developing and testing cloud infrastructure on your local host. For IaaS, you'd look at vagrant, virtual box, vmware etc. For Kubernetes you'd look at rancher etc and for PaaS you'd look at flynn, dokku, deis.

File: _posts/2018-11-12-cloud-cloud-computing.md (hunk 5 above)
Comment (zegetech) [reply _posts/2018-11-12-cloud-cloud-computing.md]: Soem open source solution for PaaS are like flynn, dokku, deis etc. Please uncover all options.

Comment (zegetech) [reply _posts/2018-11-12-cloud-cloud-computing.md]: Also please list top PaaS that are used to deploy rails application specifically.

File: _posts/2018-11-12-cloud-cloud-computing.md (hunk 1 above)
Comment (zegetech) [reply _posts/2018-11-12-cloud-cloud-computing.md]: Your intros need to be bit longer to fit the listing formats.

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+{:.post-figure}
+![The cloud](/assets/images/blog/{{page.blog-image}}){:class="img-responsive center"}
+
+
+{{page.intro}}
+
+The cloud is a pool of computer resources; servers, storage, databases, analytics, physical infrastructure and more, shared over the internet. The cloud is a virtual space that is accessible from anywhere around the globe. 
+
+The idea of the cloud is believed to have first been coined by [Joseph Carl Robnett Licklider](https://en.wikipedia.org/wiki/J._C._R._Licklider) in the 1960s.
+
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md:19]: Make this post bit more lively. e. g look for a funny cartoon about cloud to put the point across. xkcd.com has a few nice ones. Remember, this is an informative post and not practical, so in needs to be very enlightening , engaging, interesting and fun

File: _posts/2018-11-12-cloud-cloud-computing.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+The cloud itself is indestructible because like we said earlier, it is a virtual space. The cloud is not physical.
+
+## Cloud computing.
+
+cloud computing is the delivery of cloud services.
+There are other types of computing such as [Distributed Computing](https://en.wikipedia.org/wiki/Distributed_computing),[Grid computing](https://en.wikipedia.org/wiki/Grid_computing),[Utility Computing](http://en.wikipedia.org/wiki/Utility_computing), and [Cluster Computing](http://en.wikipedia.org/wiki/Computer_cluster).
+
+Cloud services are delivered in a number of ways; Infrastructure as a service(IaaS), Platform as a service(PaaS) and Software as a service(SaaS). Here is a picture of resources you would traditionally manage.
```
Comment (zegetech) [inline _posts/2018-11-12-cloud-cloud-computing.md]: Mention XaaS got from here https://searchcloudcomputing.techtarget.com/definition/XaaS-anything-as-a-service. try incorporate this as well and see the major classes of XaaS.

**Review by zegetech — CHANGES_REQUESTED** [review]
Draft review done. Cant wait to see version 2

**Comment by Melvin1Atieno** [comment]
Noted.

**Comment by Melvin1Atieno** [comment]
Version 2 is out.!!!

File: _posts/2018-11-12-cloud-cloud-computing.md (hunk 5 above)
Comment (zegetech) [reply _posts/2018-11-12-cloud-cloud-computing.md]: Please include the open source options of PaaS as i had mentioned like flynn, dokku, deis

File: _posts/2018-11-12-cloud-cloud-computing.md (hunk 8 above)
Comment (zegetech) [reply _posts/2018-11-12-cloud-cloud-computing.md]: Please add this as well.

**Comment by zegetech** [comment]
Please view the unresolved conversation and include the content.

**Review by zegetech — APPROVED** [review]

**Pull request merged**